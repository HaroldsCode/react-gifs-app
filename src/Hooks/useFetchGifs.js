import { useEffect, useState } from 'react'
import { getGifs } from "../GifExpertApp/Helpers/getGifs";

export const useFetchGifs = (category) => {
    const [state, setState] = useState({
        data: [],
        loading: true
    })

    useEffect(() => {
        getGifs(category)
            .then(imgs => {
                setState({
                    data: imgs,
                    loading: false
                })
            });    
    }, [category])

    return state;
}
