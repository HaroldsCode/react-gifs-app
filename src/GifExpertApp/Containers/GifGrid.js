import { GridItem } from "../Components/GridItem";
import PropTypes from "prop-types";
import { useFetchGifs } from "../../Hooks/useFetchGifs"

export const GifGrid = ({category}) => {
    const { data:images, loading } = useFetchGifs(category);
    return (
        <>
            <h2>Resultados de "{category}"</h2>
            <ul className="cards">
                {loading && <p>Loading Gifs...</p>}
                {images.map(({id, title, url}) => {
                    return <GridItem key={id} title={title} url={url}/>
                })}
            </ul>
        </>
    )
}

GifGrid.propTypes = {
    category: PropTypes.string.isRequired,
}