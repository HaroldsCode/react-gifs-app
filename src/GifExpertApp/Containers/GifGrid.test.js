import React from 'react';
import '@testing-library/jest-dom';
import { shallow } from "enzyme"
import { GifGrid } from "./GifGrid"
import { useFetchGifs } from '../../Hooks/useFetchGifs';
jest.mock('../../Hooks/useFetchGifs')
describe('GifGrid tests', () => {
    test('should make match with the snapshot', () => {
        useFetchGifs.mockReturnValue({
            data: [],
            loading: true
        });

        const wrapper = shallow(<GifGrid category='hola'/>);
        expect( wrapper ).toMatchSnapshot();
    })
    
    test('should display items when the images has been charged', () => {
        const gifs = [{
            id: '01',
            url: 'https://localhost/pp.jpg',
            title: 'titulito'
        }]

        useFetchGifs.mockReturnValue({
            data: gifs,
            loading: false
        });
        const wrapper = shallow(<GifGrid category='hola'/>);
        expect( wrapper.find('p').exists() ).toBe( false );
        expect( wrapper.find('GridItem').length ).toBe( gifs.length )
    })
    
})
