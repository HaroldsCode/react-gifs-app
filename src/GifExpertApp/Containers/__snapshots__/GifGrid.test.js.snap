// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`GifGrid tests should make match with the snapshot 1`] = `
<Fragment>
  <h2>
    Resultados de "
    hola
    "
  </h2>
  <ul
    className="cards"
  >
    <p>
      Loading Gifs...
    </p>
  </ul>
</Fragment>
`;
